nipype/interfaces/fsl/utils.py

```diff
@@ -26,11 +26,56 @@
 from ..base import (traits, TraitedSpec, OutputMultiPath, File,
                     CommandLine, CommandLineInputSpec, isdefined)
 from ...utils.filemanip import (load_json, save_json, split_filename,
-                                fname_presuffix)
+                                fname_presuffix, copyfile)
 
 warn = warnings.warn
 warnings.filterwarnings('always', category=UserWarning)
 
+class CopyGeomInputSpec(FSLCommandInputSpec):
+    in_file = File(exists=True, mandatory=True, argstr="%s", position=0,
+                   desc="source image")
+    dest_file = File(exists=True, mandatory=True, argstr="%s", position=1,
+                     desc="destination image", copyfile=True)
+    ignore_dims = traits.Bool(desc=('Do not copy image dimensions'),
+                              argstr='-d', position="-1")
+
+class CopyGeomOutputSpec(TraitedSpec):
+    out_file = File(exists=True, desc="image with new geometry header")
+
+class CopyGeom(FSLCommand):
+    """Use fslcpgeom to copy the header geometry information to another image.
+
+    Copy certain parts of the header information (image dimensions, voxel dimensions,
+    voxel dimensions units string, image orientation/origin or qform/sform info)
+    from one image to another. Note that only copies from Analyze to Analyze
+    or Nifti to Nifti will work properly. Copying from different files will result
+    in loss of information or potentially incorrect settings.
+
+    """
+    _cmd = "fslcpgeom"
+    input_spec = CopyGeomInputSpec
+    output_spec = CopyGeomOutputSpec
+
+    def _run_interface(self, runtime):
```

Comment:
you can replace this code with something like this - we don't like the input state getting altered in a function.

```
    def _run_interface(self, runtime):                                                                          
        # Copy destination file to new local file to prevent overwriting                                        
        # and update destination file                                                                           
        runtime = super(CopyGeom, self)._run_interface(runtime)                                                 
        out_file = self._gen_filename('out_file')                                                               
        copyfile(self.inputs.dest_file, out_file, copy=True)                                                    
        return runtime                                                                                          
                                                                                                                
    def _gen_filename(self, name):                                                                              
        if name == 'out_file':                                                                                  
            return self._gen_fname(self.inputs.dest_file, suffix="_newhd")                                      
        return None                                                                                             
                                                                                                                
    def _list_outputs(self):                                                                                    
        outputs = self.output_spec().get()                                                                      
        # Use the new destination file updated by _run_interface                                                
        outputs['out_file'] = self._gen_filename('out_file')                                                    
        return outputs                                                                                          
```